### PR TITLE
Record UI test failures & Add UI test results to report

### PR DESCRIPTION
**Pull request by sunmou99**
### Description
If test on Simulators failed with crashes, upload the test videos as artifacts.
Add new **gma_uitest** row in the test failure report.
UITest is still disabled as the failure rate is high. Will have another PR enable the UITest.

***
### Testing
Also see the comment below
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [x] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

File: scripts/gha/summarize_test_results.py [hunk 1]
```diff
@@ -378,7 +380,7 @@ def combine_configs(error_configs, all_configs):
 
 def combine_config(platform, config, config_value, k):
   config_before_combination = config.copy()
-  logging.info("platform: %s; config: %s; config_value: %s", platform, config, config_value)
+  # logging.info("platform: %s; config: %s; config_value: %s", platform, config, config_value)
```
Comment (DellaBitta) [inline scripts/gha/summarize_test_results.py:383]: nit: should this be removed completely or put back?

File: scripts/gha/test_simulator.py [hunk 2]
```diff
@@ -698,8 +742,8 @@ def _run_instrumented_test():
     "-w", "%s.test/androidx.test.runner.AndroidJUnitRunner" % CONSTANTS[FLAGS.test_type]["android_package"]] 
   logging.info("Running game-loop test: %s", " ".join(args))
   result = subprocess.run(args=args, capture_output=True, text=True, check=False) 
-  if "FAILURES!!!" in result.stdout:
-    _reset_emulator_on_error(_RESET_TYPE_REBOOT)
+  # if "FAILURES!!!" in result.stdout:
```
Comment (DellaBitta) [inline scripts/gha/test_simulator.py:745]: Here, too.

**Review by DellaBitta — CHANGES_REQUESTED** [review]
Check commented out lines.

File: scripts/gha/test_simulator.py (hunk 2 above)
Comment (sunmou99) [reply scripts/gha/test_simulator.py:745]: Let's keep the code here. The reset func is desired but not working, I will have a separate PR fixing this issue.

File: scripts/gha/summarize_test_results.py (hunk 1 above)
Comment (sunmou99) [reply scripts/gha/summarize_test_results.py]: removed

**Review by DellaBitta — APPROVED** [review]

**Comment by DellaBitta** [comment]
If test on Simulators failed with crashes, upload the test videos as artifacts.
Add new **gma_uitest** row in the test failure report.
UITest is still disabled as the failure rate is high. Will have another PR enable the UITest.

**Pull request merged**